### PR TITLE
Clean up '#if CORE' in PSUtils.cs

### PR DESCRIPTION
**Pull request by iSazonov**
Related #3565 and #4357.

### CODE REVIEW

File: src/System.Management.Automation/utils/PsUtils.cs [hunk 1]
```diff
@@ -579,15 +190,6 @@ internal static bool IsRunningOnProcessorArchitectureARM()
             {
                 return false;
             }
-#else
-            // Important:
-            // this function has a clone in Workflow.ServiceCore in admin\monad\src\m3p\product\ServiceCore\WorkflowCore\WorkflowRuntimeCompilation.cs
-            // if you are making any changes specific to this function then update the clone as well.
-
-            var sysInfo = new NativeMethods.SYSTEM_INFO();
-            NativeMethods.GetSystemInfo(ref sysInfo);
```
Comment (daxian-dbw) [inline src/System.Management.Automation/utils/PsUtils.cs]: Is `NativeMethods.SYSTEM_INFO()` still needed? If not, it should be removed as well.

Comment (iSazonov) [inline src/System.Management.Automation/utils/PsUtils.cs]: It is still used in `internal static ProcessorArchitecture GetProcessorArchitecture(out bool isRunningOnArm)` and then ` internal PSModuleInfo LoadModuleManifest()`.

Comment (daxian-dbw) [inline src/System.Management.Automation/utils/PsUtils.cs]: Then the `required processor architecture test` is broken on Unix platforms as we are depending on a win32 API to get process architecture. Can you please open an issue for this?

Comment (iSazonov) [inline src/System.Management.Automation/utils/PsUtils.cs]: IsRunningOnProcessorArchitectureARM was fixed.

Comment (iSazonov) [inline src/System.Management.Automation/utils/PsUtils.cs]: Issue #4474 is opened.

**Review by adityapatwardhan — APPROVED** [review]
LGTM

File: src/System.Management.Automation/utils/PsUtils.cs (hunk 1 above)
Comment (adityapatwardhan) [inline src/System.Management.Automation/utils/PsUtils.cs]: @daxian-dbw Can you have another look?

Comment (adityapatwardhan) [inline src/System.Management.Automation/utils/PsUtils.cs]: @daxian-dbw Do you sign off on this PR?

**Review by daxian-dbw — APPROVED** [review]

**Comment by iSazonov** [comment]
@adityapatwardhan Could you please merge?

**Pull request merged**